src/librustdoc/clean/types.rs

```diff
@@ -2377,7 +2377,7 @@ impl TypeBinding {
 crate enum SubstParam {
     Type(Type),
     Lifetime(Lifetime),
-    Constant(Constant),
+    Constant,
```

Comment:
I'd rather this field be kept since it should be used but we haven't gotten to fixing the underlying bug yet.

Reply:
Let's keep it and annotate it on the field itself with a comment stating the above (maybe all lints should accept a `reason` value?).